File: 15-yegge_questions/03-integer_sum/integer_sum.rb
# Write a function that sums up integers from a text file, one int per line.

# def sum_integers filename
#   sum = 0
#   file = File.new filename
#   file.each_line do |line|
#     sum += line.to_i
#   end
#   sum
# end

# OR

# def sum_integers filename
#   numbers = File.readlines(filename)
#   numbers.inject do |acc, num|
#     acc.to_i + num.to_i
#   end
# end

# OR

def sum_integers filename
  File.readlines(filename).map(&:to_i).inject(&:+)
end

puts sum_integers ('integers.txt')

File: 15-yegge_questions/03-integer_sum/integerSum.js
// Write a function that sums up integers from a text file, one int per line.

const fs = require('fs'); // File system code from the node std library.

const sumIntegers = function (filename) {
  let sum = 0;
  const file = fs.readFileSync(filename, 'utf-8');

  file.split('\n').forEach(function (line) {
    const n = Number(line);
    sum += n;
  });
  return sum;
};

// const sumIntegers = function (filename) {
//   const file = fs.readFileSync(filename, 'utf-8');
//   return file.split('\n').reduce(function (acc, n) {
//     return Number(acc) + Number(n);
//   });
// };

console.log( sumIntegers('./integers.txt') );


// TRY THIS WITH THE ASYNCHRONOUS VERSION
